daemon/tests/happy_path.rs

```diff
@@ -191,6 +191,43 @@ async fn force_close_an_open_cfd() {
     assert_next_state!(CfdState::Closed, maker, taker, order_id);
 }
 
+#[tokio::test]
+async fn rollover_an_open_cfd() {
+    let _guard = init_tracing();
+    let oracle_data = OliviaData::example_0();
+    let (mut maker, mut taker, order_id) =
+        start_from_open_cfd_state(oracle_data.announcement()).await;
+
+    taker.trigger_rollover(order_id).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert_eq!(taker_cfd.state, CfdState::OutgoingRollOverProposal);
+    assert_eq!(maker_cfd.state, CfdState::IncomingRollOverProposal);
+
+    maker.accept_rollover_proposal(order_id).await;
+
+    assert_next_state!(CfdState::ContractSetup, maker, taker, order_id);
+    assert_next_state!(CfdState::Open, maker, taker, order_id);
+}
```

Comment:
in a follow-up, we should really finish this test by force-closingw this cfd using a different dlc & event id, to confirm that the internals have changed (for that we'll need to extend the test framework slightly, but that should be quite easy :) )

Reply:
Agreed as discussed earlier. I'll open an issue once this merges.

Reply:
See https://github.com/itchysats/itchysats/issues/1036.